Sources/xcodes/main.swift

```diff
@@ -21,11 +26,41 @@ func loginIfNeeded() -> Promise<Void> {
     }
     .recover { error -> Promise<Void> in
         guard
-            let username = env("XCODES_USERNAME") ?? readLine(prompt: "Apple ID: "),
-            let password = env("XCODES_PASSWORD") ?? readSecureLine(prompt: "Apple ID Password: ")
+            let username = findUsername() ?? readLine(prompt: "Apple ID: "),
+            let password = findPassword() ?? readSecureLine(prompt: "Apple ID Password: ")
         else { throw Error.missingUsernameOrPassword }
 
-        return manager.client.login(accountName: username, password: password)
+        return login(username, password: password)
+    }
+}
+
+func findUsername() -> String? {
+    if let username = env(xcodesUsername) {
+        return username
+    }
+    else if let username = try? keychain.getString(xcodesUsername){
+        return username
+    }
+    return nil
+}
+
+func findPassword() -> String? {
+    if let password = env(xcodesPassword) {
+        return password
+    }
+    else if let password = try? keychain.getString(xcodesPassword){
+        return password
+    }
+    return nil
+}
+
+func login(_ username: String, password: String) -> Promise<Void> {
+    return firstly { () -> Promise<Void> in
+        manager.client.login(accountName: username, password: password)
+    }
+    .get { _ in
+        keychain["xcodesUsername"] = username
+        keychain["xcodesPassword"] = password
```

Comment:
oops `keychain["xcodesUsername"] ` shouldn't have quotes - so both the save and load with the same property name
```
let xcodesUsername = "XCODES_USERNAME"
let xcodesPassword = "XCODES_PASSWORD"
```

I'm not sure what you're meaning by 
>  think it might be more appropriate to create a single application password item with the account set to the provided Apple ID and its password set to the provided password

I was setting 2 different keychain items (username/password).  The app will first check for any environment variables (for username/password) passed in, then go to the keychain, otherwise prompt. Are you thinking some other way?

Reply:
Yes, instead of creating two keychain items and storing the username and password separately, create one and instead of setting the username to `XCODES_USERNAME`, set it it to the user's Apple ID and set the password to the password they entered.